rfc/rfc-91/rfc-91.md

```diff
@@ -130,28 +132,86 @@ We can use the same logic for preconditions with overwrite operations using the
 
 ## Test Plan
 
-We can write normal junit tests using testcontainers with GCS and S3 to simulate edge cases and general contention. Further adhoc testing will include the following scenarios:
+We can write normal junit tests using testcontainers with GCS and S3 to simulate edge cases and general contention.
 
 ### Unit tests
 
 We will add some high contention, high usage unit tests that create hundreds of threads to try and acquire locks simultaneously on the testcontainers to simulate load and contention. We can also use thread-unsafe structures like Arraylists to ensure concurrent modifications do not occur.
 
-### High-Frequency Commit and Table Service Test
 
-Run a long-running streaming ingestion process that continuously performs inserts, updates, and deletes. Ensure that frequent commits occur while table services like compaction and clustering operate concurrently. This test will help verify that the lock provider can handle overlapping operations without causing excessive delays or lock contention.
+## RFC-91-update Nov 2025: Storage LP conditional write retry handling
+
+Conditional writes with s3 which are retried due to transient issues are not idempotent, meaning that a successful PUT can go through but a retry can turn the response into a 412. This is an extremely, extremely rare case. It was not able to be reproduced, but was observed in production.
+
+To reproduce, we needed a PUT request to succeed on S3 (changing the lock owner) while the client never receives the success response (thinks it failed). However, with artifical network-level toxics between the client and S3, we face a dilemma: blocking/delaying responses either prevents the PUT from completing on S3 (timeout toxic stops everything), or the SDK's aggressive retry logic eventually succeeds after waiting through the delays (latency toxic). We can't surgically "lose" just the response after the PUT completes because TCP requires bidirectional communication for the operation to succeed on S3.
```

Comment:
For such a reproduction, we can still only block the downlink TCP data packets and allow downlink TCP ACKs to go through so that PUT request succeeds but HTTP response is blocked so the client SDK does not receive it.